### ISSUE TITLE
Segmentation metrics

### ISSUE THREAD

**Issue by emi-research-dl** [keras-team/keras-core#894]
Would it be possible to include default metrics in Keras for image segmentation? E.x Dice, IoU, HD, etc.

**Comment by emi-research-dl** [comment]
I tried this:

from keras_core import backend as K

def jaccard_distance_loss(y_true, y_pred, smooth=100):
    intersection = K.sum(K.sum(K.abs(y_true * y_pred), axis=-1))
    sum_ = K.sum(K.sum(K.abs(y_true) + K.abs(y_pred), axis=-1))
    jac = (intersection + smooth) / (sum_ - intersection + smooth)
    return (1 - jac) * smooth

def dice_metric(y_pred, y_true):
    intersection = K.sum(K.sum(K.abs(y_true * y_pred), axis=-1))
    union = K.sum(K.sum(K.abs(y_true) + K.abs(y_pred), axis=-1))
    # if y_pred.sum() == 0 and y_pred.sum() == 0:
    #     return 1.0

    return 2*intersection / union

# Example
size = 10

y_true = np.zeros(shape=(size,size))
y_true[3:6,3:6] = 1

y_pred = np.zeros(shape=(size,size))
y_pred[3:5,3:5] = 1

loss = jaccard_distance_loss(y_true,y_pred)

metric = dice_metric(y_pred,y_true)

print(f"loss: {loss}")
print(f"dice_metric: {metric}")

It seems that keras_core does not yet have backend support.
Can someone help me?

Thanks in advanced!

**Comment by soumik12345** [comment]
@emi-research-dl This is probably better suited for [KerasCV](https://github.com/keras-team/keras-cv)

**Comment by emi-research-dl** [comment]
@soumik12345  Totally agree!!
